packages/venia-concept/src/components/ProductFullDetail/ProductFullDetail.js

```diff
@@ -1,254 +1,241 @@
 import { Price } from '@magento/peregrine';
 
-import classify from 'src/classify';
+import defaultClasses from './productFullDetail.css';
+import { mergeClasses } from 'src/classify';
+
 import Button from 'src/components/Button';
 import { loadingIndicator } from 'src/components/LoadingIndicator';
 import Carousel from 'src/components/ProductImageCarousel';
 import Quantity from 'src/components/ProductQuantity';
 import RichText from 'src/components/RichText';
-import defaultClasses from './productFullDetail.css';
+
 import appendOptionsToPayload from 'src/util/appendOptionsToPayload';
 import findMatchingVariant from 'src/util/findMatchingProductVariant';
 import isProductConfigurable from 'src/util/isProductConfigurable';
 
 const Options = React.lazy(() => import('../ProductOptions'));
 
-class ProductFullDetail extends Component {
-    static propTypes = {
-        classes: shape({
-            cartActions: string,
-            description: string,
-            descriptionTitle: string,
-            details: string,
-            detailsTitle: string,
-            imageCarousel: string,
-            options: string,
-            productName: string,
-            productPrice: string,
-            quantity: string,
-            quantityTitle: string,
-            root: string,
-            title: string
-        }),
-        product: shape({
-            __typename: string,
-            id: number,
-            sku: string.isRequired,
-            price: shape({
-                regularPrice: shape({
-                    amount: shape({
-                        currency: string.isRequired,
-                        value: number.isRequired
-                    })
-                }).isRequired
-            }).isRequired,
-            media_gallery_entries: arrayOf(
-                shape({
-                    label: string,
-                    position: number,
-                    disabled: bool,
-                    file: string.isRequired
-                })
-            ),
-            description: string
-        }).isRequired,
-        addToCart: func.isRequired
-    };
-
-    static getDerivedStateFromProps(props, state) {
-        const { configurable_options } = props.product;
-        const optionCodes = new Map(state.optionCodes);
+const INITIAL_OPTION_CODES = new Map();
+const INITIAL_OPTION_SELECTIONS = new Map();
+const INITIAL_QUANTITY = 1;
 
-        // if this is a simple product, do nothing
-        if (!isProductConfigurable(props.product)) {
-            return null;
-        }
-
-        // otherwise, cache attribute codes to avoid lookup cost later
-        for (const option of configurable_options) {
-            optionCodes.set(option.attribute_id, option.attribute_code);
-        }
-
-        return { optionCodes };
+const deriveOptionCodesFromProduct = product => {
+    // If this is a simple product it has no option codes.
+    if (!isProductConfigurable(product)) {
+        return INITIAL_OPTION_CODES;
```

Comment:
I can't make this change because there are other downstream functions that expect `optionCodes` to be a `Map`, even if it is an empty one.

One example is [findMatchingVariant](https://github.com/magento-research/pwa-studio/blob/develop/packages/venia-concept/src/util/findMatchingProductVariant.js#L9).